File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.java
class Solution {
    public int strStr(String haystack, String needle) {
       int h=haystack.length(),n=needle.length();
        for(int i=0;i<=h-n;i++){
            String str=haystack.substring(i,i+n);
            if(str.equals(needle)) return i;
        }
        return -1;
    }
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
  let h = haystack.length;
  let n = needle.length;
  let maxIdxToCheck = h - n;

    for (let i = 0; i <= maxIdxToCheck; i++) {
      let subString = haystack.substring(i, i + n)
      if (subString === needle) {
        return i;
      }
    }
    return -1;
};